src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -1180,41 +1292,15 @@ public void updateApplicationMetadata(UUID guid, org.cloudfoundry.client.v3.Meta
     @Override
     public void updateApplicationStaging(String applicationName, Staging staging) {
         UUID applicationGuid = getRequiredApplicationGuid(applicationName);
-        UpdateApplicationRequest.Builder requestBuilder = UpdateApplicationRequest.builder();
-        requestBuilder.applicationId(applicationGuid.toString());
-        if (staging != null) {
-            requestBuilder.buildpack(staging.getBuildpack())
-                          .command(staging.getCommand())
-                          .healthCheckHttpEndpoint(staging.getHealthCheckHttpEndpoint())
-                          .healthCheckTimeout(staging.getHealthCheckTimeout())
-                          .healthCheckType(staging.getHealthCheckType())
-                          .enableSsh(staging.isSshEnabled());
-            String stackName = staging.getStack();
-            if (stackName != null) {
-                UUID stackGuid = getStack(stackName).getMetadata()
-                                                    .getGuid();
-                requestBuilder.stackId(stackGuid.toString());
-            }
-            if (staging.getDockerInfo() != null) {
-                requestBuilder.dockerImage(staging.getDockerInfo()
-                                                  .getImage());
-                DockerCredentials dockerCredentials = staging.getDockerInfo()
-                                                             .getCredentials();
-                if (dockerCredentials != null) {
-                    requestBuilder.dockerCredentials(org.cloudfoundry.client.v2.applications.DockerCredentials.builder()
-                                                                                                              .username(dockerCredentials.getUsername())
-                                                                                                              .password(dockerCredentials.getPassword())
-                                                                                                              .build());
-                }
-            }
-        }
-        delegate.applicationsV2()
-                .update(requestBuilder.build())
+        UpdateApplicationRequest updateApplicationRequest = UpdateApplicationRequest.builder()
```

Comment:
Why is this not inlined in the `update` method?

Reply:
This method is used to update the application. CreateOrUpdateStep -> update strategy